esp-hal/src/dma/gdma.rs

```diff
@@ -91,6 +121,14 @@ use embassy_sync::waitqueue::AtomicWaker;
 static TX_WAKERS: [AtomicWaker; CHANNEL_COUNT] = [const { AtomicWaker::new() }; CHANNEL_COUNT];
 static RX_WAKERS: [AtomicWaker; CHANNEL_COUNT] = [const { AtomicWaker::new() }; CHANNEL_COUNT];
 
+cfg_if::cfg_if! {
+    if #[cfg(any(esp32c2, esp32c3))] {
+        use portable_atomic::AtomicBool;
+        static TX_IS_ASYNC: [AtomicBool; CHANNEL_COUNT] = [const { AtomicBool::new(false) }; CHANNEL_COUNT];
```

Comment:
I'm guessing the PR that makes use of these are WIP. I'm currently questioning why it's atomic and not `UnsafeCell`.

Reply:
Yeah, I think I rebased away where I checked these bits. They don't do much yet - not until we actually allow fake-splitting channels.

`AtomicBool` because why not, I'm hoping codegen gives us atomic load/store.